packages/layers/auth/tests/unit/composables/useAuthApi.test.ts

```diff
@@ -0,0 +1,149 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { describe, it, expect, vi, beforeEach } from 'vitest'
+import { mockNuxtImport } from '@nuxt/test-utils/runtime'
+
+const mockAuthApi = vi.fn()
+mockNuxtImport('useNuxtApp', () => {
+  return () => ({
+    $authApi: mockAuthApi
+  })
+})
+
+const mockOpenErrorModal = vi.fn()
+mockNuxtImport('useConnectTosModals', () => {
+  return () => ({
+    openPatchTosErrorModal: mockOpenErrorModal
+  })
+})
+
+const mockQuery = vi.fn()
+const mockInvalidateQueries = vi.fn()
+
+const { mockDefineQuery, mockUseMutation } = vi.hoisted(() => {
+  const mockMutationReturnObject = {
+    mutateAsync: vi.fn(),
+    isPending: false,
+    isSuccess: false
+  }
+
+  const mockUseMutation = vi.fn(() => mockMutationReturnObject)
+
+  const mockDefineQuery = vi.fn(() => mockQuery)
+
+  return {
+    mockDefineQuery,
+    mockUseMutation
+  }
+})
+
+vi.mock('@pinia/colada', () => ({
+  defineQuery: mockDefineQuery,
+  useQueryCache: () => ({ invalidateQueries: mockInvalidateQueries }),
+  useMutation: mockUseMutation,
+  defineMutation: vi.fn(setupFn => setupFn)
+}))
+
+describe('useAuthApi', () => {
+  const authApi = useAuthApi()
+
+  beforeEach(() => {
+    vi.clearAllMocks()
+  })
+
+  describe('getAuthUserProfile', () => {
+    it('should define and return the query', async () => {
+      const expected = { data: 'test-data' }
+      mockQuery.mockReturnValue(expected)
+
+      const result = await authApi.getAuthUserProfile()
+
+      expect(mockDefineQuery).toHaveBeenCalledTimes(1)
+
+      const callArgs = (mockDefineQuery.mock.calls[0] as any)[0]
+
+      expect(callArgs.key).toEqual(['auth-user-profile'])
+      expect(callArgs.staleTime).toBe(300000)
+
+      expect(mockQuery).toHaveBeenCalledTimes(1)
+
+      expect(result).toBe(expected)
+
+      const queryFunction = callArgs.query
+      queryFunction()
+
+      expect(mockAuthApi).toHaveBeenCalledWith('/users/@me', {
+        parseResponse: JSON.parse
+      })
+    })
+  })
+
+  describe('getTermsOfUse', () => {
+    it('should define and return the query', async () => {
+      const expected = { data: 'test-data' }
+      mockQuery.mockReturnValue(expected)
+
+      const result = await authApi.getTermsOfUse()
+
+      expect(mockDefineQuery).toHaveBeenCalledTimes(1)
+
+      const callArgs = (mockDefineQuery.mock.calls[0] as any)[0]
+
+      expect(callArgs.key).toEqual(['auth-terms-of-use'])
+      expect(callArgs.staleTime).toBe(300000)
```

Comment:
Do we care about testing the pinia colada args? We're setting a precedent here where we want every query/mutation checked in a unit test for the library interaction. When we upgrade and the interaction changes or if we switch libraries they will all fail and its a lot of test changes for something of pretty low value in my opinion since we are already testing everything around it